backend/server.js

```diff
@@ -0,0 +1,224 @@
+const express = require("express");
+const cors = require("cors");
+const multer = require("multer");
+
+const upload = multer();
+
+const cookieParser = require("cookie-parser");
+const app = express();
+
+const MongoClient = require("mongodb").MongoClient;
+
+const passwordCheck = require(__dirname + "/utilities/passwordCheck.js");
+const generateRandomString = require(__dirname +
+  "/utilities/generateRandomString.js");
+const passwordHash = require("password-hash");
+
+// app.listen(4000, () => {
+//    // LOCAL SERVER
+//    console.log("Running on port 4000");
+// });
+//_____________________MOGODB Setup___________________
+
+let dataBaseUrl =
+  "mongodb+srv://admin:WhqtQsNqxt@cluster0-58fms.mongodb.net/test?retryWrites=true&w=majority";
+
+let UsersDB;
+let Collection_LoginInfo;
+let Collection_Sessions;
+
+(async function initDB() {
+  await MongoClient.connect(
+    dataBaseUrl,
+    { useNewUrlParser: true },
+    (err, allDbs) => {
+      console.log(
+        "-----------------------Database Initialised-----------------------"
+      );
+      // Add option useNewUrlParser to get rid of console warning message
+      if (err) throw err;
+      UsersDB = allDbs.db("Users");
+      Collection_LoginInfo = UsersDB.collection("LoginInfo");
+      Collection_Sessions = UsersDB.collection("Sessions");
+    }
+  );
+})();
+
+//_____________________MIDLEWARE_______________________
+app.use(cookieParser());
+app.use(cors());
+app.use(cors({ credentials: true, origin: "http://localhost:3000" })); // CONFIG FOR LOCAL SERVER
+app.use("/images", express.static(__dirname + "/uploads")); // Files in local folder uploads have endpoints as /images/x
+app.use("/assets", express.static(__dirname + "/assets"));
+
+//app.use(cors({ credentials: true, origin: "http://134.209.119.133:3000" })); // CONFIG FOR REMOTE SERVER
+
+//_________________Begining of END POINTS____________________
+
+app.get("/", (req, res) => {
+  res.send("hello");
+});
+//----------------------------------------------------------------------------
+
+//SIGNUP
+app.post("/signup", upload.none(), (req, res) => {
+  if (req.body === undefined) {
+    console.log("no data with login request");
+    res.status(400);
+    res.send(
+      JSON.stringify({ success: false, result: "signup data not recieved" })
```

Comment:
Nit: there's a shorthand for this, you can use `res.json` instead:

```js
res.json({ success: false, result: "signup data not received" });
```

You can also chain the status method:

```js
res.status(400).json({ success: false, result: "signup data not received" });
```

Also: "Nit" is a common convention in code-reviews. It means slightly different things to different people, but broadly it means "this is a nitpick suggestion that shouldn't block merging. I'd appreciate if you made this tweak, but it's non-critical and if you're in a hurry you can ignore it"